src/file.ts

```diff
@@ -4081,7 +4084,13 @@ class File extends ServiceObject<File> {
  * that a callback is omitted.
  */
 promisifyAll(File, {
-  exclude: ['publicUrl', 'request', 'save', 'setEncryptionKey'],
+  exclude: [
+    'publicUrl',
+    'request',
+    'save',
+    'setEncryptionKey',
+    'shouldRetryBasedOnPreconditionAndIdempotencyStrat',
```

Comment:
might have to update unit tests because of this

Reply:
Yes, it broke a bunch of other tests as well which I am trying to figure out.